cpp/include/cuspatial/point_linestring_nearest_points.hpp

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cudf/column/column_view.hpp>
+#include <cudf/utilities/span.hpp>
+
+#include <optional>
+
+namespace cuspatial {
+
+/**
+ * @ingroup nearest_points
+ * @struct point_linestring_nearest_points_result
+ * @brief Container for the result of `pairwise_point_linestring_nearest_points`
+ *
+ * @var point_linestring_nearest_points_result::nearest_point_geometry_id
+ * The point id indicating which point in the multipoint is the nearest point
+ * (`std::nullopt` if input is not a multipoint array)
+ *
+ * @var point_linestring_nearest_points_result::nearest_linestring_geometry_id
+ * The linestring id indicating which linestring in the multilinestring that
+ * the nearest point locates (`std::nullopt` if input is not a multlinestring array)
+ *
+ * @var point_linestring_nearest_points_result::nearest_segment_id
+ * The segment id indicating which segment in the linestring that nearest point locates.
+ * It is the same as the id to the starting point of the segment. Each linestring in the
+ * multilinestring is independently indexed.
+ *
+ * @var point_linestring_nearest_points_result::nearest_point_on_linestring_xy
+ * The interleaved x, y-coordinate of the nearest point on the (multi)linestring
+ */
+struct point_linestring_nearest_points_result {
+  std::optional<std::unique_ptr<cudf::column>> nearest_point_geometry_id;
+  std::optional<std::unique_ptr<cudf::column>> nearest_linestring_geometry_id;
+  std::unique_ptr<cudf::column> nearest_segment_id;
+  std::unique_ptr<cudf::column> nearest_point_on_linestring_xy;
+};
```

Comment:
Why not document the members of the struct inline and eliminate the @struct, @var, etc?

```suggestion
struct point_linestring_nearest_points_result {
  /// The point id indicating which point in the multipoint is the nearest point
  /// (`std::nullopt` if input is not a multipoint array)
  std::optional<std::unique_ptr<cudf::column>> nearest_point_geometry_id;
  
  /// etc.
  std::optional<std::unique_ptr<cudf::column>> nearest_linestring_geometry_id;
  
  /// etc.
  std::unique_ptr<cudf::column> nearest_segment_id;
  std::unique_ptr<cudf::column> nearest_point_on_linestring_xy;
};
```

Reply:
This is better. Thanks.